python/lsst/sphgeom/interval1d.cc

```diff
@@ -0,0 +1,67 @@
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#include "pybind11/pybind11.h"
+
+#include <memory>
+#include <stdexcept>
+
+#include "lsst/sphgeom/Interval1d.h"
+#include "lsst/sphgeom/python/interval.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace sphgeom {
+namespace {
+
+PYBIND11_PLUGIN(interval1d) {
+    py::module mod("interval1d");
+
+    py::class_<Interval1d, std::shared_ptr<Interval1d>> cls(mod, "Interval1d");
+
+    python::defineInterval<decltype(cls), Interval1d, double>(cls);
+
+    cls.def_static("empty", &Interval1d::empty);
+    cls.def_static("full", &Interval1d::full);
+
+    cls.def(py::init<>());
+    cls.def(py::init<double>(), "x"_a);
+    cls.def(py::init<double, double>(), "x"_a, "y"_a);
+    cls.def(py::init<Interval1d const &>(), "interval"_a);
+
+    cls.def("isFull", &Interval1d::isFull);
+
+    cls.def("__str__", [](Interval1d const &self) {
+        return py::str("[{!s}, {!s}]").format(self.getA(), self.getB());
+    });
+    cls.def("__repr__", [](Interval1d const &self) {
+        return py::str("Interval1d({!r}, {!r})")
+                .format(self.getA(), self.getB());
+    });
```

Comment:
Yes it can, as can all the other interval-like classes you mention - see `include/lsst/sphgeom/interval.h`. There is no base class exposed to Python; instead, that header adds lambdas that call the various interval methods, including one for `__reduce__`. This works for the interval types with a CRTP base class, as well as `NormalizedAngleInterval`.